clients/src/main/java/org/apache/kafka/common/requests/FetchRequest.java

```diff
@@ -492,74 +327,51 @@ public int maxBytes() {
     }
 
     public boolean isFromFollower() {
-        return replicaId >= 0;
+        return replicaId() >= 0;
     }
 
     public IsolationLevel isolationLevel() {
-        return isolationLevel;
+        return IsolationLevel.forId(data.isolationLevel());
     }
 
     public FetchMetadata metadata() {
         return metadata;
     }
 
     public String rackId() {
-        return rackId;
+        return data.rackId();
     }
 
     public static FetchRequest parse(ByteBuffer buffer, short version) {
-        return new FetchRequest(ApiKeys.FETCH.parseRequest(version, buffer), version);
+        ByteBufferAccessor accessor = new ByteBufferAccessor(buffer);
```

Comment:
In the parsing logic, we still convert to struct first before calling `AbstractRequest.parseRequest`. I think we could bypass the `Struct` conversion by changing `AbstractRequest.parseRequest` to take the `ByteBuffer` instead of the `Struct`. 
```java
    public static AbstractRequest parseRequest(ApiKeys apiKey, short apiVersion, ByteBuffer buffer) {
```
Then in the fetch case, we could just call this method.

Reply:
I believe this is also addressed in @ijuma's PR.